pkg/migrations/sql/idempotent/base.sql

```diff
@@ -1429,39 +1486,76 @@ COMMENT ON FUNCTION SCHEMA_CATALOG.epoch_abort(BIGINT)
 IS 'ABORT an INSERT transaction due to the ID epoch being out of date';
 GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.epoch_abort TO prom_writer;
 
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.get_confirmed_unused_series(
+    series_table TEXT, potential_series_ids BIGINT[]
+) RETURNS BIGINT[]
+AS $func$
+DECLARE
+    r RECORD;
+BEGIN
+    FOR r IN
+        SELECT *
+        FROM SCHEMA_CATALOG.metric m
+        WHERE m.series_table = get_confirmed_unused_series.series_table
+    LOOP
+
+    --at each iteration of the loop filter potential_series_ids to only
+    --have those series ids that don't exist in the metric tables.
+    EXECUTE format(
+    $query$
+        SELECT array_agg(potential_series.series_id)
+        FROM unnest($1) as potential_series(series_id)
+        LEFT JOIN LATERAL(
+            SELECT 1
+            FROM  %1$I.%2$I data_exists
+            WHERE data_exists.series_id = potential_series.series_id
+            --use chunk append + more likely to find something starting at earliest time
+            ORDER BY time ASC
+            LIMIT 1
+        ) as lateral_exists(indicator) ON (true)
+        WHERE lateral_exists.indicator IS NULL
+    $query$, r.table_schema, r.table_name)
+		USING potential_series_ids
+    INTO potential_series_ids;
+
+    END LOOP;
+    RETURN potential_series_ids;
+END
+$func$
+LANGUAGE PLPGSQL VOLATILE
+--security definer to add jobs as the logged-in user
+SECURITY DEFINER
+--search path must be set for security definer
+SET search_path = pg_temp;
+--redundant given schema settings but extra caution for security definers
+REVOKE ALL ON FUNCTION SCHEMA_CATALOG.get_confirmed_unused_series(TEXT, BIGINT[]) FROM PUBLIC;
+GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.get_confirmed_unused_series(TEXT, BIGINT[]) TO prom_maintenance;
+
 CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.mark_unused_series(
     metric_table TEXT, older_than TIMESTAMPTZ, check_time TIMESTAMPTZ
 ) RETURNS VOID AS $func$
+DECLARE
 BEGIN
     --chances are that the hour after the drop point will have the most similar
     --series to what is dropped, so first filter by all series that have been dropped
     --but that aren't in that first hour and then make sure they aren't in the dataset
     EXECUTE format(
     $query$
         WITH potentially_drop_series AS (
-            SELECT distinct series_id
-            FROM SCHEMA_DATA.%1$I
-            WHERE time < %2$L
+            SELECT distinct id as series_id
+            FROM SCHEMA_DATA_SERIES.%1$I
```

Comment:
ugh I don't think this approach will work because the logic here was to take all the series ids that were in the "section to be deleted" and diff them "EXCEPT" the first hour after deletion to get a small set. The change of getting all series in the table and diffing with the first hour will still yield a huge set. I suggest we switch back to the approach of passing the check_time down to the get_confirmed_unused_series but only apply it to the table we are dropping from and not the other tables/views.